### PR TITLE
[ML] Inference document configurable settings

### PR DESCRIPTION
**Pull request by jonathan-buttner**
This PR documents the adjustable inference API settings.

I wouldn't expect a user to adjust the settings. It's more to document what's available in case support needs to tweak things.

I took a stab at creating these, I'm happy to reorganize them if they don't match how we typical do these docs. I just wanted to get the information down since we hadn't documented it yet.

### CODE REVIEW

File: docs/reference/settings/inference-settings.asciidoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+[role="xpack"]
+[[inference-settings]]
+=== Inference API settings in Elasticsearch
+++++
+<titleabbrev>Inference settings</titleabbrev>
+++++
+
+[[inference-settings-description]]
+// tag::inference-settings-description-tag[]
+You do not need to configure any settings to use the {infer} APIs. Each setting has a default set.
```
Comment (maxhniebergall) [inline docs/reference/settings/inference-settings.asciidoc:11]:
```suggestion
You do not need to configure any settings to use the {infer} APIs. Each setting has a default.
```

File: docs/reference/settings/inference-settings.asciidoc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+[[xpack-inference-input-text]]
+// tag::inference-input-text[]
+==== Inference API Input text
+
+For certain 3rd party service integrations, when the service returns an error indicating that the request
+input was too large, the input will be truncated and the request is retried. These settings govern
+how the truncation is performed.
+
+`xpack.inference.truncator.reduction_percentage`::
```
Comment (jonathan-buttner) [inline docs/reference/settings/inference-settings.asciidoc:89]: This isn't actually the name of this setting. There was a typo that is being corrected in this PR: https://github.com/elastic/elasticsearch/pull/108219/files#diff-24707283cf310e9531dd805fdbaf557c7847f2cc63642ccc8d3fe8d5e80741d3R29

**Review by maxhniebergall — APPROVED** [review]
LGTM!

**Review by szabosteve — APPROVED** [review]
LGTM, thanks for documenting this. I left a couple of suggestions but only nits. Please take or leave them.

**Pull request merged**